File: backend/app/schemas/assistant.py
from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.ambulances import Ambulance
from app.schemas.blood import BloodBank
from app.schemas.hospitals import Hospital
from app.schemas.schemes import EligibleScheme


Severity = Literal["Critical", "Moderate", "Low"]


class EmergencyPlanRequest(BaseModel):
    message: str = Field(min_length=3, max_length=2000)
    patient_age: int = Field(ge=0, le=120)
    location: str = Field(min_length=2, max_length=200)


class EmergencyPlanResponse(BaseModel):
    detected_emergency: str
    severity: Severity
    triage_summary: str
    recommended_action: str
    recommended_hospital: Hospital | None
    ambulance_suggestion: Ambulance | None
    blood_bank_suggestion: BloodBank | None
    scheme_suggestions: list[EligibleScheme]
    step_by_step_plan: list[str]
    disclaimer: str

File: frontend/lib/assistant.ts
import type { Ambulance } from "@/lib/ambulances";
import { apiClient } from "@/lib/api-client";
import type { BloodBank } from "@/lib/blood-bank";
import type { Hospital } from "@/lib/hospitals";
import type { EligibleScheme } from "@/lib/schemes";

export type EmergencyPlanRequest = {
  message: string;
  patient_age: number;
  location: string;
};

export type EmergencyPlanResponse = {
  detected_emergency: string;
  severity: "Critical" | "Moderate" | "Low";
  triage_summary: string;
  recommended_action: string;
  recommended_hospital: Hospital | null;
  ambulance_suggestion: Ambulance | null;
  blood_bank_suggestion: BloodBank | null;
  scheme_suggestions: EligibleScheme[];
  step_by_step_plan: string[];
  disclaimer: string;
};

export async function generateEmergencyPlan(
  payload: EmergencyPlanRequest,
): Promise<EmergencyPlanResponse> {
  return apiClient<EmergencyPlanResponse>("/api/assistant/emergency-plan", {
    errorMessage: "Unable to generate emergency plan",
    method: "POST",
    body: payload,
  });
}
